### DISCUSSION TITLE
Ability to set the Step for Queries

### DISCUSSION THREAD

**Discussion by RECturtle** [grafana/grafana#59885] [Dashboards, Panels & Visualizations]
We've got a datasource that is scraping a source every 5m for metrics. Specifically we're retrieving a metric that has total count of new docs indexed in the past 5 minutes (not a running total, just how many new docs from now to now-5minutes). Because of this, we've set the min interval to 5m so our graphs and calculation on this data in grafana are accurate (e.g. sum_over_time/reduce Total). The issue we're running into is if I want to view the statistics using a 7 day timerange for a dashboard, the interval jumps to 10m and we end up undersampling our data and the stats are incorrect.

Ideally, we'd have a way to set a static step so we don't return incorrect data and Grafana would return an error if we were attempting to retrieve too many data points from a prometheus source (I know the previous implementation of setting the step ran into this issue). For us, the error message would be much better than not having control of the step.

Otherwise, does anyone have any recommendations on what to do? My only thought currently is to add additional custom metrics that have higher time samples (e.g. past 1 hour) and then set the min interval to greater than 1 hour since it is unlikely we query for a range that would cause the interval to jump higher than that

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). Due to the age and number of responses to this discussion, we are deciding to close it. If this is something you would like to see in Grafana, feel free to open an [issue](https://github.com/grafana/grafana/issues) so the discussion can continue. Thank you!
